hollow/src/main/java/com/netflix/hollow/api/producer/WriteStateImpl.java

```diff
@@ -26,9 +28,12 @@
  * @author Tim Taylor {@literal<tim@toolbear.io>}
  */
 final class WriteStateImpl implements HollowProducer.WriteState {
+    private static final String LATE_MODIFICATION_MESSAGE =
+            "attempt to modify state after populate stage complete; version=%d";
     private final long version;
-    private final HollowObjectMapper objectMapper;
-    private final HollowProducer.ReadState priorReadState;
+    private HollowObjectMapper objectMapper;
+    private HollowProducer.ReadState priorReadState;
+    private volatile boolean sealed = false;
 
     protected WriteStateImpl(long version, HollowObjectMapper objectMapper, HollowProducer.ReadState priorReadState) {
```

Comment:
Give WriteStateImpl is a package protected final class the protected modified is redundant.